### DISCUSSION TITLE
Floating-point stack with respect to CATCH / THROW.

### DISCUSSION THREAD

**Discussion by SirWumpus** [ForthHub/discussion#135] [Q&A] [answered]
The Forth Draft 19-1 section 12 Floating-point word set makes not reference as to how a separate float stack is handle with respect to `CATCH` and `THROW`.  I assume, that for a separate float stack, the float stack pointer is also saved and restored by `CATCH` and `THROW`.

* Would that be a fair assumption?
* Is this something that should be addressed by the next Draft?

**Comment by ruv** [comment, marked as answer]
The section [9.6.1.2275 THROW](https://forth-standard.org/standard/exception/THROW) says:
- > adjust the depths of **all stacks defined by this standard** so that they are the same as the depths saved in the exception frame

One small problem (as I pointed out in my [comment](https://forth-standard.org/proposals/specify-that-0-throw-pops-the-0#reply-857)) is that saving of depths is not explicitly specified for `catch`, I think by an omission.

So, yes, if an exception occurs, the floating-point stack depth shall be restored.
